katharsis-core/src/main/java/io/katharsis/core/internal/resource/ReflectionFieldAccessor.java

```diff
@@ -0,0 +1,91 @@
+package io.katharsis.core.internal.resource;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+import io.katharsis.core.internal.utils.ClassUtils;
+import io.katharsis.core.internal.utils.PropertyException;
+import io.katharsis.core.internal.utils.PropertyUtils;
+import io.katharsis.resource.information.ResourceFieldAccessor;
+
+public class ReflectionFieldAccessor implements ResourceFieldAccessor {
+
+	private Method getter;
+
+	private Method setter;
+
+	private Field field;
+
+	private Class<?> resourceType;
+
+	private String fieldName;
+
+	private Class<?> fieldType;
+
+	public ReflectionFieldAccessor(Class<?> resourceType, String fieldName, Class<?> fieldType) {
+		if (fieldName == null) {
+			throw new IllegalArgumentException("no fieldName provided");
+		}
+		if (resourceType == null) {
+			throw new IllegalArgumentException("no resourceType provided");
+		}
+		if (fieldType == null) {
+			throw new IllegalArgumentException("no fieldType provided");
+		}
+		this.resourceType = resourceType;
+		this.fieldName = fieldName;
+		this.fieldType = fieldType;
+		this.getter = ClassUtils.findGetter(resourceType, fieldName);
+		this.setter = ClassUtils.findSetter(resourceType, fieldName, fieldType);
+		this.field = ClassUtils.findClassField(resourceType, fieldName);
+		if (field != null && !Modifier.isPublic(field.getModifiers())) {
+			this.field = null;
+		}
+
+	}
+
+	@Override
+	public Object getValue(Object resource) {
+		if (resource == null) {
+			String message = String.format("Cannot get value %s.%s for null", resourceType.getCanonicalName(), fieldName);
+			throw new PropertyException(message, resourceType, fieldName);
+		}
+		try {
+			if (field != null) {
+				return field.get(resource);
```

Comment:
Although it is rarely an issues, you should always do 
```
boolean current = field.getAccessible();
field.setAccessible(true);
// do stuff
field.setAccessible(current);
```

Because, if the security manager is engaged this will fail with `IllegalAccessException` - Dumb, I know, but it happens based on the env it runs in that we have no control over